From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 8 Jan 2021 02:14:42 +0100
Subject: workspacesView: Fix off-by-one error

The scroll adjustment's upper value corresponds to the number of
workspaces, not to the last workspace index. We want the latter
when mirroring the layout in RTL locales, so subtract 1.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1598>
(cherry picked from commit 9f0e7632a6e1cb94bb9bf71002d548a8cbb2d558)
Origin: upstream, 3.38.4, commit:07ef8c57a4372a01b6b0ac583cf1e89c479ae3f9
---
 js/ui/workspacesView.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/workspacesView.js b/js/ui/workspacesView.js
index 2cc8b04..eba43ad 100644
--- a/js/ui/workspacesView.js
+++ b/js/ui/workspacesView.js
@@ -292,7 +292,7 @@ class WorkspacesView extends WorkspacesViewBase {
         const vertical = workspaceManager.layout_rows === -1;
         const rtl = this.text_direction === Clutter.TextDirection.RTL;
         const progress = vertical || !rtl
-            ? adj.value : adj.upper - adj.value;
+            ? adj.value : adj.upper - adj.value - 1;
 
         for (const ws of this._workspaces) {
             if (vertical)
